solver/src/main/java/org/chocosolver/solver/expression/continuous/arithmetic/BiCArExpression.java

```diff
@@ -276,4 +281,8 @@ public void intersect(double lb, double ub, ICause cause) throws ContradictionEx
     public String toString() {
         return op.name() + "(" + e1.toString() + "," + e2.toString() + ")";
     }
+
+    private boolean isIntegerConstant(RealVar realVar) {
+        return realVar.isAConstant() && (realVar.getLB() % 1) == 0;
```

Comment:
What about using this:
`&& Math.rint(realVar.getLB()) == realVar.getLB()`
Regarding the `rint(a)`Javadoc:
> Special cases:
     If the argument value is already equal to a mathematical integer, then the result is the same as the argument.

Reply:
Sure, no problem. I've commited the change